src/js/createTippy.js

```diff
@@ -476,20 +476,23 @@ export default function createTippy(reference, collectionProps) {
    * Creates the popper instance for the tip
    */
   function createPopperInstance() {
-    const { tooltip } = tip.popperChildren
     const { popperOptions } = tip.props
+    const { tooltip, arrow } = tip.popperChildren
 
-    const arrowSelector =
-      Selectors[tip.props.arrowType === 'round' ? 'ROUND_ARROW' : 'ARROW']
-    const arrow = tooltip.querySelector(arrowSelector)
-
-    const config = {
+    return new Popper(tip.reference, tip.popper, {
       placement: tip.props.placement,
       ...(popperOptions || {}),
       modifiers: {
         ...(popperOptions ? popperOptions.modifiers : {}),
+        preventOverflow: {
+          boundariesElement: tip.props.boundary,
+          ...(popperOptions && popperOptions.modifiers
+            ? popperOptions.modifiers.preventOverflow
+            : {}),
```

Comment:
Maybe you can create a helper for extracting `popperOptions`? I can see this patterns a few times here.

Reply:
Good point